src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -571,16 +573,26 @@ DeferErrorIfCannotPushdownSubquery(Query *subqueryTree, bool outerQueryHasLimit)
 		errorDetail = "Offset clause is currently unsupported";
 	}
 
-	if (subqueryTree->limitCount && !outerQueryHasLimit)
+	/*
+	 * Limit is partially supported when SubqueryPushdown is set.
+	 * The outermost query must have a limit clause.
+	 */
+	if (subqueryTree->limitCount && !SubqueryPushdown)
```

Comment:
Also, in the master branch, we error out `LIMIT`s whenever there is no LIMIT on the outside query. But, here we skip that if the flag is set?

Is that an informed decision? Thinking about, this change makes more sense to me. What are your thoughtson this?